jax/numpy/lax_numpy.py

```diff
@@ -3375,7 +3380,8 @@ def _index_to_gather(x_shape, idx):
         # XLA gives error when indexing into an axis of size 0
         raise IndexError(f"index is out of bounds for axis {x_axis} with size 0")
       i = _normalize_index(i, x_shape[x_axis])
-      i = lax.convert_element_type(i, index_dtype)
+      # dummy index if is polynomial, doesn't matter for shape inference:
+      i = 0 if type(i) is Poly else lax.convert_element_type(i, index_dtype)
```

Comment:
This seems pretty subtle... hmm...

Reply:
I'm going to leave a todo here to follow up and figure out the best way to do this.

Reply:
`i = lax.convert_element_type(i, index_dtype)` doesn't work for cases like `x[-1]` where `i` becomes a `Poly` representing `n-1`, breaking `convert_element_type`.

Let me know if you can come up with a better solution!